src/libs/sys-util-rs/src/mount.rs

```diff
@@ -0,0 +1,808 @@
+    if idx <= 1 {
+        return ("".to_string(), n_opts);
+    }
+
+    let idx = idx as usize;
+    let common_dir = get_longest_common_prefix(&lower_opts);
+    if common_dir == "/" {
+        return ("".to_string(), n_opts);
+    }
+
+    info!(
+        sl!(),
+        "compact_lowerdir_option get common prefix: {}", &common_dir
+    );
+    let lower: Vec<String> = lower_opts
+        .iter()
+        .map(|c| c.replace(common_dir.as_str(), ""))
+        .collect();
+    n_opts[idx as usize] = format!("lowerdir={}", lower.join(":"));
+    (common_dir, n_opts)
+}
+
+fn find_overlay_lowerdirs(opts: &[String]) -> (isize, Vec<String>) {
+    for (idx, o) in opts.iter().enumerate() {
+        if let Some(lower) = o.strip_prefix("lowerdir=") {
+            if !lower.is_empty() {
+                let c_opts: Vec<String> = lower.split(':').map(|c| c.to_string()).collect();
+                return (idx as isize, c_opts);
+            }
+        }
+    }
+
+    (-1, Vec::new())
+}
+
+// assume lower always with same prefix
+fn get_longest_common_prefix(opts: &[String]) -> String {
+    //let paths: Vec<Path> = opts.iter().map(|v| Path::new(v)).collect();
+    if opts.is_empty() {
+        return String::new();
+    }
+    if opts.len() == 1 {
+        return opts[0].clone();
+    }
+
+    // FIXME: get prefix
+    let strs = opts[0].clone();
+    let split_l: Vec<&str> = strs.split('/').collect();
+
+    //let mut prefix = &split_l[..1].join("/");
+    let mut prefix = "/".to_string();
+    let mut idx: usize = 0;
+    for i in 1..split_l.len() {
+        let tmp_l = &split_l[..i];
+        let tmp_ll = tmp_l.join("/");
+        if !tmp_ll.is_empty() {
+            prefix = tmp_ll.to_string();
+        }
+
+        let contain = || -> bool {
+            for opt in opts.iter() {
+                if !opt.contains(&prefix) {
+                    return false;
+                }
+            }
+            true
+        };
+
+        if !contain() {
+            break;
+        }
+        idx = i;
+    }
+
+    let tmp_l = &split_l[..idx];
+    let tmp_ll = tmp_l.join("/");
+    prefix = tmp_ll + "/";
+
+    prefix
+}
+
+/// Umount a mountpoint with timeout.
+pub fn umount_timeout(path: &str, timeout: u64) -> Result<()> {
+    let parent = Path::new(path)
+        .parent()
+        .ok_or_else(|| Error::InvalidPath(path.to_owned()))?;
+    let parent_meta = fs::metadata(&parent).map_err(|e| Error::ReadMetadata(path.to_owned(), e))?;
+    let meta = fs::metadata(path).map_err(|e| Error::ReadMetadata(path.to_owned(), e))?;
+    if meta.file_type().is_symlink() || parent_meta.file_type().is_symlink() {
+        warn!(sl!(), "unable to umount {} which is a symbol link", &path);
+        return Ok(());
+    }
+
+    if timeout == 0 {
+        nix::mount::umount2(path, nix::mount::MntFlags::MNT_DETACH)
+            .map_err(|e| Error::Umount(path.to_owned(), e))?;
+        info!(sl!(), "lazy umount for {}", path);
```

Comment:
How is this "lazy"?

Reply:
umount2(MNT_DETACH) means lazy umount:
```
       MNT_DETACH (since Linux 2.4.11)
              Perform a lazy unmount: make the mount unavailable for new
              accesses, immediately disconnect the filesystem and all
              filesystems mounted below it from each other and from the
              mount table, and actually perform the unmount when the
              mount ceases to be busy.
```

Reply:
Ok, understood.